src/transformers/generation/logits_process.py

```diff
@@ -361,6 +361,31 @@ def __call__(self, input_ids: torch.LongTensor, scores: torch.FloatTensor) -> to
         if self.prompt_ignore_length:
             input_ids = input_ids[:, self.prompt_ignore_length :]
 
+        # Ensure input_ids is int64 for gather operation
+        input_ids = input_ids.to(torch.int64)
+
+        # Handle continuous batching case where scores have shape [batch_size, seq_len, vocab_size]
+        if scores.dim() == 3:
+            # For continuous batching, apply repetition penalty only to the last position
+            # This is compatible with CUDA graphs as it avoids dynamic operations
+            batch_size, seq_len, vocab_size = scores.shape
+
+            last_scores = scores[:, -1, :]  # [batch_size, vocab_size]
```

Comment:
not sure this works, last spot would just be one of the last tokens, as batch is always 1.